homeassistant/components/soma/cover.py

```diff
@@ -21,39 +36,139 @@ async def async_setup_entry(
     """Set up the Soma cover platform."""
 
     devices = hass.data[DOMAIN][DEVICES]
+    entities = []
+
+    for device in devices:
+        # Assume a shade device if the type is not present in the api response (Connect <2.2.6)
+        if "type" in device and device["type"].lower() == "tilt":
+            entities.append(SomaTilt(device, hass.data[DOMAIN][API]))
+        else:
+            entities.append(SomaShade(device, hass.data[DOMAIN][API]))
```

Comment:
If you set `api = hass.data[DOMAIN][API]` above this would be a bit cleaner

Reply:
fixed